src/d3a/models/area/__init__.py

```diff
@@ -374,26 +380,33 @@ def publish_market_cycle_to_external_clients(self):
 
     def _consume_commands_from_aggregator(self):
         if self.redis_ext_conn is not None and self.redis_ext_conn.is_aggregator_controlled:
-            self.redis_ext_conn.aggregator.\
-                consume_all_area_commands(self.uuid,
-                                          self.redis_ext_conn.trigger_aggregator_commands)
-        elif self.strategy is not None \
-                and hasattr(self.strategy, "is_aggregator_controlled") \
-                and self.strategy.is_aggregator_controlled:
-            self.strategy.redis.aggregator.\
-                consume_all_area_commands(self.uuid,
-                                          self.strategy.trigger_aggregator_commands)
+            (self.redis_ext_conn.aggregator.
+             consume_all_area_commands(self.uuid,
+                                       self.redis_ext_conn.trigger_aggregator_commands))
+        elif (self.strategy
+              and hasattr(self.strategy, "is_aggregator_controlled")
+              and self.strategy.is_aggregator_controlled):
+            (self.strategy.redis.aggregator.
+             consume_all_area_commands(self.uuid,
+                                       self.strategy.trigger_aggregator_commands))
 
     def tick(self):
+        """Tick event handler.
+
+        invokes aggregator commands consumer, publishes market clearing, updates events,
+        updates cached market's bids and offers in case of myco matching and matches
+        bid offer pairs otherwise.
+        """
         self._consume_commands_from_aggregator()
 
-        if ConstSettings.IAASettings.MARKET_TYPE == 2:
+        if (ConstSettings.IAASettings.MARKET_TYPE != SpotMarketTypeEnum.ONE_SIDED.value
+                and not self.strategy):
             if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
                 self.dispatcher.publish_market_clearing()
             else:
                 if is_external_matching_enabled():
-                    bid_offer_matcher.match_algorithm.area_uuid_markets_mapping.\
-                        update({self.uuid: self.all_markets})
+                    bid_offer_matcher.match_algorithm.update_area_uuid_markets_mapping(
+                        {self.uuid: self.all_markets})
                 else:
                     for market in self.all_markets:
                         bid_offer_pairs = bid_offer_matcher.calculate_recommendation(
```

Comment:
This comment is probably hardly addressed in this PR, since it's a bit broad. However, it is not a "Nit" comment, in the sense that I believe it is quite important for the future.

I think that there's something off with this part (conceptually).
We implemented a `MycoMatcher` class that should act as an interface for all different types of matching. This means that we should call the same methods for all different types of `MycoMatcher`.
However, we're not using it in that way: here we are differentiating between the `MycoMatcher`s whose `match_algorithm` is external, and the other `MycoMatcher`s. In fact, calling this line:
```python
bid_offer_matcher.match_algorithm.update_area_uuid_markets_mapping
```
only works for `ExternalMatcher` instances, and will break for `PayAsClearMatcher` or `PayAsBidMatcher`. I think this defeats the purpose of using interfaces (the `BaseMatcher`).

Conceptually, we should not directly call methods on the `match_algorithm` here. We should have some interface in `bid_offer_matcher` (the `MycoMatcher`) that deals with it, so that we can remove the `if is_external_matching_enabled()` check from here.

Reply:
To be a bit clearer, what I would expect is (as a first step) something like this:
```python
def tick(self):
    self._consume_commands_from_aggregator()

    if (ConstSettings.IAASettings.MARKET_TYPE != SpotMarketTypeEnum.ONE_SIDED.value
            and not self.strategy):
        if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
            self.dispatcher.publish_market_clearing()
        else:
        	bid_offer_matcher.my_interface_method()
```

Reply:
@fievelk I see your points, however I think there is a confusion here, we should distinguish between a matcher and a matching algorithm
I think this is the root issue, kindly note that there is no BaseMatcher anymore in the d3a, there is a BaseMatchingAlgorithm in the d3a-interface, this is the `interface` for all matching algorithms and they will all have the same interface `get_matches_recommendations`
Most probably we need to rename either `MycoMatcher` or `ExternalMatcher`, the matcher suffix should not be on both
Unfortunately we cannot homogenize anymore here because the external matcher is the one that requests offers and bids , we cannot explicitly call `match_recommendations` on it, he is the one to fire events unlike our default setup
Please note that the ExternalMatcher could make use of PayAsBidMatchingAlgorithm, PayAsClearMatchingAlgorithm, AttributedMatchingAlgorithm or any other client defined algorithm
I agree about the last point (my_interface_method), I'll move it to the MycoMatcher, and btw we should pass data as well so if external matching is not enabled we should match on the go.

Reply:
> Unfortunately we cannot homogenize anymore here because the external matcher is the one that requests offers and bids , we cannot explicitly call match_recommendations on it, he is the one to fire events unlike our default setup

I think this is exactly the issue here. The `bid_offer_matcher` (`MycoMatcher`) can contain components whose behaviour is incompatible (I'm talking in terms of interface). That's why you need that `if` in front of it: you can't call the same interface method because it won't work for all components. I think this points out the fact that something could be changed in the structure of the classes. In some cases the `MycoMatcher` is just contacting an API, while in some other cases it's actually computing recommendations (and this create an incoherent structure IMO).

Reply:
@fievelk why are you considering the Myco Matcher as an interface ?
this is the class that is aware of the different matching algorithms and their different interfaces (the externalMatcher vs all other matching algorithms)
this can be solved when we (if ever) be in charge of sending the open offers/bids to the myco and wait for its recommendations, however I suggested this to Colin and he seemed positive but I still believe that this is to be thought more of.